h2integrate/core/utilities.py

```diff
@@ -729,3 +729,144 @@ def check_file_format_for_csv_generator(
     new_file.close()
 
     return new_fpath
+
+
+def print_results(model, includes=None, excludes=None, show_units=True):
+    """Print hierarchical inputs plus explicit/implicit outputs (means only) using Rich.
+
+    Order of rows preserves OpenMDAO's original ordering from list_inputs/list_outputs.
+    Group rows are emitted lazily the first time a variable within that path appears.
+    """
+
+    def _gather_outputs(explicit=True, implicit=False):
+        return model.list_outputs(
+            explicit=explicit,
+            implicit=implicit,
+            val=True,
+            prom_name=True,
+            units=show_units,
+            shape=True,
+            includes=includes,
+            excludes=excludes,
+            out_stream=None,
+            return_format="list",
+        )
+
+    explicit_meta = _gather_outputs(explicit=True, implicit=False)
+    implicit_meta = _gather_outputs(explicit=False, implicit=True)
+
+    # Gather inputs (no explicit/implicit split in OpenMDAO API)
+    input_meta = model.list_inputs(
+        val=True,
+        prom_name=True,
+        units=show_units,
+        shape=True,
+        includes=includes,
+        excludes=excludes,
+        out_stream=None,
+        return_format="list",
+    )
+
+    def _mean(val):
+        if isinstance(val, np.ndarray):
+            return "nan" if val.size == 0 else f"{np.mean(val)}"
+        if isinstance(val, (int, float, np.number)):
+            return f"{val}"
+        return "n/a"
+
+    from rich import box
+    from rich.table import Table
+    from rich.console import Console
+
+    console = Console()
+
+    def _emit_section(title, meta_list, kind_label="outputs"):
+        if not meta_list:
+            return
+        console.print(f"\n{len(meta_list)} {title.lower()} {kind_label}:")
+        table = Table(show_header=True, header_style="bold", box=box.MINIMAL, pad_edge=False)
+        table.add_column("Variable", overflow="fold")
+        table.add_column("Mean")
```

Comment:
```suggestion
        table.add_column("Mean", justify="right")
```
This will make more sense in my next comment, but I noticed the float printing is quite difficult to read, so I'm adding this to help.